functions-and-scope.js

```diff
@@ -42,43 +69,89 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 
 // ---- Verwachte uitkomst: 6.642857142857143
 
+// Gemiddelde = alle waarden bij elkaar optellen / het aantal waarden dus in dit geval [i]/grades.size
+// Alle lossen waarden moeten aangeroepen worden
+// (let i = 0; i <= grades.length; i++)
+
+let total = 0;
+
+for (let i = 0; i < grades.length; i++) {
+    total = total + grades[i];
+}
+console.log(total / grades.length);
 
 /* 2b: Omschrijven tot een herbruikbare functie */
-// Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
-// Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde
+// cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken,
+// zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
 
 // ---- Verwachte uitkomsten:
 // averageGrade(grades) geeft 6.642857142857143
 // averageGrade([6, 4, 5]) geeft 5
 // averageGrade([8, 9, 4, 6, 10]) geeft 7.4
 
+function averageGrade(gradesArray) {
+    let total = 0;
+
+    for (let i = 0; i < gradesArray.length; i++) {
+        total = total + gradesArray[i];
+    }
+
+    return total / gradesArray.length;
+}
+
+const gradesOne = averageGrade(grades);
+const gradesTwo = averageGrade([6, 4, 5]);
+const gradesThree = averageGrade([8, 9, 4, 6, 10]);
+
+console.log(gradesOne, gradesTwo, gradesThree);
 
 /* 2c: Afronden op twee decimalen */
 // Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
 // Tip: Google is your best friend!
 
+function averageRound(num) {
+    return Math.round((num + Number.EPSILON) * 100) / 100;
```

Comment:
Keurig. De toFixed() methode zou nog korter zijn. Die retourneerd echter wel een string datatype, wat in dit geval natuurlijk prima is. Mocht je er later nog berekeningen op willen toepassen, dan is jouw manier beter aangezien die een Number datatype retourneerd.